docsp-34315 - 1.13 version updates (#317)
mongodb/docs-golang: source/whats-new.txt

   :depth: 1
   :class: singlecol

.. facet::
   :name: genre
   :values: reference
 
.. meta::

>    :keywords: update, backward compatibility, version

Reviewer: S: omit version

* :ref:`Version 1.1 <version-1.1>`
* :ref:`Version 1.0 <version-1.0>`

.. _version-1.13:

What's New in 1.13
------------------

The 1.13 {+driver-short+} release includes the following improvements and fixes:

- Logging for server selection and SDAM. See the :ref:`golang-logging` documentation

>   for more information about logging.

Author: changed but it's weird to me that we use "guide" when we've also been planning to create a specific "Guides" section of the docs. leaving as "documentation" but lmk if you want it changed

>   :ref:`golang-logging` documentation for more information about logging.

Reviewer: S: I think there are too many clauses in sentence 2

Author: doh! good catch

> - Methods on the ``mongo.Collection`` type for managing search indexes programmatically.

Reviewer: ```suggestion
- Methods on the ``Collection`` type that allow you to manage search indexes programmatically.
```

- The ``event.CommandStartedEvent`` and ``event.CommandFinishedEvent`` events return
  the ``DatabaseName`` field. This field is also included in the command-logging analogs.

> - In a sharded topology, when selecting a server to retry a failing query, the driver

Reviewer: vale fix
```suggestion
- In a sharded topology, when selecting a server to retry an unsuccesful query, the driver
```

Author: so silly to me that vale complains about this.

Reviewer 2: @mongoKart 

Just to provide context, this is directly from [Negative Words](https://www.mongodb.com/docs/meta/style-guide/writing/use-positive-statements/#avoid-negative-words) guidance in the style guide. I think it's reasonable guidance, but can understand that it's a subjective rule.

Here's my suggestion:
If you want to identify specific cases in which we should allow negative words, can you include those exceptions in a style guide PR? That way, other team reps can weigh in. Based on the results, we can update the Vale rule.
Thanks!

>   excludes the server used for the initial failed attempt. Instead, if there's more than

Reviewer: ```suggestion
  excludes the server used for the initial attempt. Instead, if there's more than
```

  one eligible ``mongos`` instance, the driver randomly selects one. Unhealthy instances
  are automatically excluded from selection. 

> - Streaming SDAM is disabled by default on AWS Lambda and similar function-as-a-service
>   (FaaS) platforms. You can change the monitoring mode by using the
>   ``serverMonitoringMode`` URI option.

Reviewer: S: to avoid passive voice and vale error, what if you reworded as follows:
```suggestion
- You can enable streaming SDAM events on AWS Lambda and similar function-as-a-service
  (FaaS) platforms by changing the monitoring mode. You can change the monitoring mode by setting
  the ``serverMonitoringMode`` URI option to....
```

Author: - seems relevant to mention that it's disabled by default on these platforms? especially if users are assuming it's going to be enabled.
- i intentionally omitted the enabling code from this section, figuring we'd put it elsewhere in the docs. do you think it's particularly helpful here?

Reviewer: I dont think that level of detail is necessary for the whats new section.
